server/mscalendar/notification_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package mscalendar
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"golang.org/x/oauth2"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/config"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/mscalendar/mock_plugin_api"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/remote/mock_remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store/mock_store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot/mock_bot"
+)
+
+func NewTestNotificationProcessor(env Env) NotificationProcessor {
+	processor := &notificationProcessor{
+		Env: env,
+	}
+	return processor
+}
+
+func TestProcessNotification(t *testing.T) {
+	token := &oauth2.Token{
+		AccessToken: "creator_oauth_token",
+	}
+
+	eRemote := &remote.Event{
```

Comment:
Do you think each of these test structs can be moved to functions such as `newTestEvent`?